common/lib/xmodule/xmodule/html_module.py

```diff
@@ -76,6 +76,17 @@ def student_view(self, _context):
         """
         return Fragment(self.get_html())
 
+    def student_view_data(self):
+        """
+        Returns a JSON representation of the student_view of this XBlock,
+        retrievable from the Course Block API.
+        Since this data is not user-specific, fields like `html` may contain
+        placeholders like %%USER_ID%%.
```

Comment:
For your use-cases, wouldn't you want to replace `%%USER_ID%%` with the requesting user's ID when called via the Course Blocks API?  You can do so by implementing a BlockTransformer and updating all HTML block data in the `transform` method.

Reply:
@nasthagiri Good point. Do you know how important that `%%USER_ID%%` us / how it is used? It's weird to me that the html module has code for `%%USER_ID%%` rather than using [this method|https://github.com/edx/edx-platform/blob/e8a36957b1f732974260e7b9b42b9c25148b492c/common/djangoapps/util/keyword_substitution.py#L34-L57] which substitutes that placeholder as well as others like `%%USER_FULLNAME%%`.

Do you consider that block transformer a requirement for merging this PR?

Reply:
@bradenmacdonald No, I don't see this as a blocker for merging this PR.